.changeset/quiet-suits-count.md

```diff
@@ -0,0 +1,57 @@
+---
+'@ithaka/pharos': major
+'@ithaka/pharos-cli': minor
+'@ithaka/pharos-site': minor
+---
+
+require manual component registration:
+
+* Remove component self-registration
+* Scope registries for components composed of other Pharos components using [scoped-registry-mixin](https://github.com/lit/lit/tree/main/packages/labs/scoped-registry-mixin)
+* Update React wrapper to detect defined custom tag name
+* Create `PharosElement` base class which sets a custom attribute `data-pharos-component` as a common identifier
+* Set components to extend `PharosElement`
+* Update documentation
+
+**Migration Guidelines**
+
+1. Remove all individual imports of Pharos web components.
+
+2. Register them with a custom tag in your app's entrypoint in the form of `{app/bundle}-pharos-{component}`:
+
+    ```javascript
+    import { PharosAlert } from '@ithaka/pharos/lib/components/alert/pharos-alert';
+
+    customElements.define('homepage-pharos-alert', PharosAlert);
+    ```
```

Comment:
We could also provide a little helper (_at some point_) that could make it easier for folks who need to register a big bunch of components:

```es6
import { registerComponents } from '@ithaka/pharos/lib/utils';

import { PharosAlert } from ...
import { PharosDropdownMenu } from ...

// spits out <homepage-pharos-alert>, <homepage-pharos-dropdown-menu>, ...
registerComponents('homepage', [ PharosAlert, PharosDropdownMenu, ...])
```

The consumer could do the following, I suppose? Not sure how fun `component-name-converted-to-kebab-case` will be at first glance 🤷🏻‍♂️ 

```es6
import { PharosAlert } from ...
import { PharosDropdownMenu } from ...

[ PharosAlert, PharosDropdownMenu, ...].forEach((component) => {
    customElements.define(`homepage-${<component-name-converted-to-kebab-case>}`, component);
});
```

Reply:
The first approach does feel nice and could help minimize human error